### PR TITLE
Rename KEYPAIR to KEY_PAIR and psa_generate_random_key to psa_generate_key

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
`KEY_PAIR` is consistent with `PUBLIC_KEY`.

“Generate key” is a more classical name. The longer name was only introduced to avoid confusion with getting a key from a generator, which is key derivation, but we no longer use the generator terminology so this reason no longer applies.

This PR is built on top of #112, only the last two commits are new.

### CODE REVIEW

**Comment by Patater** [comment]
I reviewed only the last two commits in this PR.

I've checked that there is no remaining KEYPAIR or psa_generate_random_key text present. All text has been changed, and the text that did change has changed correctly.

File: include/psa/crypto_extra.h [hunk 1]
```diff
@@ -436,7 +436,7 @@ psa_status_t psa_generate_derived_key_to_handle(psa_key_handle_t handle,
                                       size_t bits,
                                       psa_key_derivation_operation_t *operation);
 
-psa_status_t psa_generate_random_key_to_handle(psa_key_handle_t handle,
+psa_status_t psa_generate_key_to_handle(psa_key_handle_t handle,
```
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply include/psa/crypto_extra.h]: We never released a version where it was called `psa_generate_random_key` and it had these arguments. So this function actually shouldn't have been renamed from `psa_generate_key` to `psa_generate_key_to_handle`.

I think I made a similar mistake with key derivation. We can fix that later, after we've decided what we actually want for the transition period. I only picked these names to have the functions around while I was migrating the tests.

**Comment by Patater** [comment]
CI failure is ABI job (known to fail until https://github.com/ARMmbed/mbedtls/pull/2636 lands in the development branch) and `USE_PSA_CRYPTO`, which doesn't work on the API branch.

**Comment by gilles-peskine-arm** [comment]
I just pushed identical content, rebased on top of the merged of #112.

**Comment by Patater** [comment]
Trivial rebase. No changes after rebase, so no re-testing required.

**Pull request merged**